datalad/cmd.py

```diff
@@ -98,6 +98,221 @@ def _cleanup_output(stream, std):
+          as a byte-string to this callable, in the chunks it was received
+          by polling the process (see `poll_latency`).
+          The callable may transform it in any way. It must
+          return a byte-string of the transformed output, and an integer
+          with the number of bytes at the end of the original output
+          that were left unprocessed (or 0, if the entire output was
+          considered). The returned byte-strings are concatenated and
+          provided as stdout return value.
+          The helper functions 'kill_output' and 'capture_output' are
+          provided to either swallow all output (and not relay it to the
+          parent) or to capture all output and provide it as the return
+          value.
+        proc_stderr : callable, optional
+          Like proc_stdout, but for stderr.
+        stdin : byte stream, optional
+          File descriptor like, used as stdin for the process. Passed
+          verbatim to subprocess.Popen().
+        poll_latency : float, optional
+          Shortest interval at which the running process is queried for
+          output (in seconds). Any potential output processing will increase
+          the effective interval. When the effective polling frequency is
+          too low to keep the output buffers below their maximum size,
+          a process will deadlock.
+
+        Returns
+        -------
+        stdout, stderr
+          Unicode string with the cumulative standard output and error
+          of the process.
+
+        Raises
+        ------
+        CommandError
+          On execution failure (non-zero exit code) this exception is
+          raised which provides the command (cmd), stdout, stderr,
+          exit code (status), and a message identifying the failed
+          command, as properties.
+        """
+        proc_out = (proc_stdout, proc_stderr)
+        try:
+            lgr.log(8, "Start running %r", cmd)
+            process = subprocess.Popen(
+                cmd,
+                stdout=subprocess.PIPE if proc_stdout else None,
+                stderr=subprocess.PIPE if proc_stderr else None,
+                shell=False,
+                cwd=self.cwd,
+                env=self.env,
+                stdin=stdin,
+                # intermediate reports are never decoded anyways
+                # from PY37 onwards
+                #text=False,
+                universal_newlines=False,
+            )
+        except Exception as e:
+            lgr.log(11, "Failed to start %r%r: %s" %
+                    (cmd,
+                     (" under %r" % self.cwd) if self.cwd else '',
+                     exc_str(e)))
+            raise
+
+        try:
+            out = [[], []]
+            unprocessed = [None, None]
+            # make sure to run this loop at least once, even if the
+            # process is already dead
+            keep_going = True
+
+            def _handle_output(u, o, p):
```

Comment:
`p` isn't used in the function.  Instead `proc` is taken from the outer scope.

Reply:
:man_facepalming: 

Fixed.